conductor/lib/file_utils.py

```diff
@@ -364,61 +364,55 @@ def quote_path(filepath):
     return '"%s"' % filepath.replace('"', '\\"')
 
 
-def reconstruct_filename(matched, file_pieces):
-    full_file_string = ""
-    matched = matched.lower()
-    for count, piece in enumerate(file_pieces):
-
-        full_file_string += piece
-
-        # If there's more to the file peices add the frame number
-        if len(file_pieces) > count + 1:
-            dummy_piece = piece.lower()
-            frame_val = matched.split(dummy_piece)[1]
-            frame_val = frame_val.split(file_pieces[count + 1].lower())[0]
-            full_file_string += frame_val
-
-    return full_file_string
-
-
-def get_files_from_path_expression(path_expression, no_extension=False):
+def get_files_from_path_expression(path_expression):
     '''
     Given a path expression (such as an image sequence path), seek out all files
     that are part of that path expression (e.g all of the files that are part
     of that image sequence) and return a list of their explicit paths.
     This function relies on what is actually on disk, so only files that are
     found on disk will be returned. If no files are found, return an empty list.
 
-    Supports a variaty of path expressions. Here are a few examples:
+    Supports a variety of path expressions. Here are a few examples:
         "image.####.exr"   # Hash syntax
         "image.####"       # no extension  - if there is no extension for the file then that must be specified by the no_extension argument
         "image.%04d.exr"   # printf format
         "image<UDIM>.exr   # Udim
         "image.$F.exr      # Houdini
+
+    In addition to matching against the file name/root, this function also matches expressions
+    found against the directory name/path, e.g.
+        /data/shot-###/image.exr
+        /data/shot-###/image.####.exr
+        /data/shot-###/image.%04d.exr
+        /data/shot-###/camera-$F/image.%04d.exr
     '''
-    dirpath, basename, extension = separate_path(path_expression, no_extension=no_extension)
+
     logger.debug("Evaluating path expression: %s", path_expression)
 
-    rx = get_rx_match(basename, PATH_EXPRESSIONS)
-    if rx:
+    # Cycle through all regexes. Note that a single path may contain more than one expression format,
+    # therefore we must test all expressions (no limit).
+    rxs = get_rx_matches(path_expression, PATH_EXPRESSIONS, limit=0)
```

Comment:
`limit` already defaults to `0` in the function's signature.

Reply:
The `get_rx_matches` function could have easily defaulted to a limit of `1` (it's what I first wrote actually).  I wanted to be very explicit about the limit (hence the big comment about it), because it's not necessarily expected behavior, and it may change In the future.  The reason being is that when having to cycle through each regex, we may find that this function doesn't scale so well when running on hundreds of thousands of files.  That was the thinking anyway.

Regardless, I made the change as you suggested